sw/device/tests/aes_interrupt_encryption_test.c

```diff
@@ -0,0 +1,261 @@
+           sizeof(kAesModesPlainTextBlock12));
+
+    // Encrypt kAesNumBlocks blocks.
+    TRY(dif_aes_process_data(&aes, plain_text, cipher_text,
+                             (size_t)kAesNumBlocks));
+    LOG_INFO("AES Encrypt of first two blocks done");
+
+    dif_aes_iv_t iv_encrypt1;
+    TRY(dif_aes_read_iv(&aes, &iv_encrypt1));
+
+    bool ready = false;
+
+    // Disable the AES unit to no longer automatically start
+    // encryption/decryption
+
+    transaction.manual_operation = kDifAesManualOperationManual;
+
+    TRY(dif_aes_trigger(&aes, kDifAesTriggerDataOutClear));
+    TRY(dif_aes_trigger(&aes, kDifAesTriggerKeyIvDataInClear));
+
+    LOG_INFO("De-initialized Automatic operation");
+
+    //***************  Perform an trial encryption using a different data and
+    // key. Message size, key size and mode can be chosen arbitrarily.
+
+    TRY(dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes));
+
+    // Mask the key. Note that this should not be done manually. Software is
+    // expected to get the key in two shares right from the beginning.
+
+    uint8_t key_share0_t[sizeof(kAesModesKey256)];
+    for (int i = 0; i < sizeof(kAesModesKey256); ++i) {
+      key_share0_t[i] = kAesModesKey256[i] ^ kKeyShare1[i];
+    }
+
+    // "Convert" key share byte arrays to `dif_aes_key_share_t`.
+    dif_aes_key_share_t key_t;
+    memcpy(key_t.share0, key_share0, sizeof(key_t.share0));
+    memcpy(key_t.share1, kKeyShare1, sizeof(key_t.share1));
+
+    dif_aes_iv_t iv2;
+    memcpy(iv2.iv, kAesModesIvCtr, sizeof(iv2.iv));
+
+    dif_aes_transaction_t transaction_trial = {
+        .operation = kDifAesOperationEncrypt,
+        .mode = kDifAesModeCtr,
+        .key_len = kDifAesKey256,
+        .key_provider = kDifAesKeySoftwareProvided,
+        .mask_reseeding = kDifAesReseedPerBlock,
+        .manual_operation = kDifAesManualOperationAuto,
+        .reseed_on_key_change = false,
+        .ctrl_aux_lock = false,
+    };
+
+    TRY(dif_aes_start(&aes, &transaction_trial, &key_t, &iv2));
+
+    // "Convert" plain data byte arrays to `dif_aes_data_t` array.
+    enum {
+      kAesNumBlocks_t2 = 4,
+    };
+
+    dif_aes_data_t plain_text2[kAesNumBlocks_t2];
+    memcpy(plain_text2[0].data, kAesModesPlainText, sizeof(kAesModesPlainText));
+    TRY(dif_aes_load_data(&aes, plain_text2[0]));
+
+    do {
+      TRY(dif_aes_get_status(&aes, kDifAesStatusOutputValid, &ready));
+    } while (!ready);
+
+    //*****************Trial encryption ends ******************
+    //  Not checking successful encryption here since its a trial transaction
+    //  Trial encryption ends here
+    //  Resume the Original Encryption below
+
+    // Using the iv read from `iv_encrypt1` for resuming the encryption of
+    // remaining 2 blocks.
+
+    memcpy(iv.iv, &iv_encrypt1, sizeof(iv.iv));
+    transaction.manual_operation = kDifAesManualOperationAuto;
+
+    // "Convert" key share byte arrays to `dif_aes_key_share_t`.
```

Comment:
I think this is not necessary as `key` should still contain the key used for the first two blocks?